### PR TITLE
Add batchedReduceAnd Op to Glow

### PR DESCRIPTION
**Pull request by sgeorgiev255**
Summary:
Add batchedReduceAnd op which performs a reduction operation similar to batchedReduce(Add/Min/Max).

Test Plan:
Relevant unit tests have been added to OperatorTest.

### CODE REVIEW

File: lib/Backends/Interpreter/InterpreterNodes.cpp [hunk 1]
```diff
@@ -4180,6 +4180,49 @@ void BoundInterpreterFunction::fwdBatchedReduceAddInst(
...
+        for (dim_t w = 0; w < eBatchDims[3]; w++) {
+          for (dim_t q = 0; q < eBatchDims[4]; q++) {
+            for (dim_t r = 0; r < eBatchDims[5]; r++) {
+              dim_t destIndices[] = {x, y, z, w, q, r};
+              if (destIndices[axis] == 0) {
+                eDestH.at(destIndices) = true;
+              }
+              destIndices[axis] = 0;
+              eDestH.at(destIndices) =
+                  eDestH.at(destIndices) && eBatchH.at({x, y, z, w, q, r});
```
Comment (jfix71) [inline lib/Backends/Interpreter/InterpreterNodes.cpp:4217]: So I think we have very similar logic for the other reduction cases. I'm wondering if we could take an opportunity to try to cleanly unify them somehow? E.g. it was partially done in `DEFINE_REDUCEADDPROD_INST_IMPL`. Any thoughts on how easy it'd be?

File: lib/Graph/Nodes.cpp [hunk 2]
```diff
@@ -1531,6 +1531,7 @@ bool LengthsSumNode::verify() const {
   }
 
 DEFINE_BATCHED_REDUCTION_VERIFICATION(BatchedReduceAdd)
+DEFINE_BATCHED_REDUCTION_VERIFICATION(BatchedReduceAnd)
```
Comment (jfix71) [inline lib/Graph/Nodes.cpp:1534]: In this case we only ever expect to support Bool input/outputs right? Maybe we can also verify that is the case?

**Review by jfix71 — COMMENTED** [review]
Seems sane -- mostly wondering about reusing similar looping logic in the interpreter implementation across the different reduce instructions.

File: lib/Backends/Interpreter/InterpreterNodes.cpp (hunk 1 above)
Comment (dangus36) [reply lib/Backends/Interpreter/InterpreterNodes.cpp]: I've made changes to make use of the same logic. Would this be a sufficient solution?

File: lib/Graph/Nodes.cpp (hunk 2 above)
Comment (dangus36) [reply lib/Graph/Nodes.cpp]: I've removed the macro and made it's own verification function with the boolean check

**Pull request closed without merging**